app/code/Magento/Inventory/Indexer/Scope/UnknownStateException.php

```diff
@@ -0,0 +1,18 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Inventory\Indexer\Scope;
+
+use Magento\Framework\Exception\LocalizedException;
+
+/**
+ * Exception for situation where used state which is not defined in configuration
+ * @todo refactoring it copy from catalog search module
+ * @api
+ */
+class UnknownStateException extends LocalizedException
```

Comment:
Btw, I don't know why we have this Exception in our codebase. 
So, from what I see we have a pre-defined finite list of states which could be switched in pre-defined order. (aka [State machine design pattern](https://en.wikipedia.org/wiki/State_pattern)) 

- our state object initially being initialized in 'REGULAR' state.
- after that, we've decided to move it to 'TEMPORARY' state while making re-indexation
- after re-indexation complete - we move the object to 'REGULAR' state again.

Is it correct? I suppose so, then why do we have a possibility that State object would be created in a wrong state? 
@larsroettig